sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java

```diff
@@ -207,16 +207,24 @@ private AnalyzedRelation normalSelect(QueriedTable table,
                 tnp.outputs(allOutputs);
                 collectNode = PlanNodeBuilder.collect(tableInfo, whereClause, toCollect, ImmutableList.<Projection>of(tnp));
 
+                // MERGE
+                tnp = new TopNProjection(limit, querySpec.offset());
+                tnp.outputs(finalOutputs);
                 if (orderBy == null) {
-                    tnp = new TopNProjection(limit, querySpec.offset());
+                    // no sorting needed
+                    mergeNode = PlanNodeBuilder.localMerge(ImmutableList.<Projection>of(tnp), collectNode);
                 } else {
-                    tnp = new TopNProjection(limit, querySpec.offset(),
-                            orderByInputColumns,
+                    // no order by needed in TopN as we already sorted on collector
+                    // and we merge sorted with SortedBucketMerger
+                    int[] orderByIndices = OrderByPositionVisitor.process(orderByInputColumns, allOutputs);
+                    mergeNode = PlanNodeBuilder.sortedLocalMerge(
```

Comment:
can the sortedLocalMerge method take orderBy and then use the OrderByPositionVisitor? Seems like the OrderByPositionVisitor is only used for the sortedLocalMerge